trie/hexary.py

```diff
@@ -131,20 +153,88 @@ def _get(self, root_hash, trie_key):
         else:
             raise Exception("Invariant: This shouldn't ever happen")
 
-    def _traverse(self, root_hash, trie_key) -> Tuple['Node', Tuple[int, ...]]:
+    def traverse(self, trie_key_input: Nibbles) -> TrieNode:
+        """
+        Find the node at the path of nibbles provided. Unspecified behavior if
+        path ends in the middle of an extension.
+
+        :raises MissingTraversalNode: if a node body is missing from the database
+        """
+        trie_key = Nibbles(trie_key_input)
+
+        node, remaining_key = self._traverse(self.root_hash, trie_key)
+
+        if len(remaining_key) > 0:
+            path_to_node = trie_key[:len(trie_key) - len(remaining_key)]
+            annotated_node = self._annotate_node(node)
+            raise TraversedPartialPath(path_to_node, annotated_node)
+        else:
+            return self._annotate_node(node)
+
+    @classmethod
```

Comment:
this could be a static method or even a standalone util function?  I don't see any access to `cls`